examples/typing/mg.jl

```diff
@@ -3,6 +3,149 @@ using Cassette
 using LightGraphs
 using MetaGraphs
 
+Cassette.@context TypeCtx
+            
+"""   TypeCtx
+
+creates a MetaDiGraph tracking the types of args and ret values throughout a script
+
+"""
+TypeCtx
+            
+            
+
+"""   FCollector(depth::Int,frame::function,data::FCollector)
+
+struct to collect all the "frames" called throughout a script
+        
+"""
+mutable struct FCollector{I,F,C}
+    depth::I
+    frame::F
+    data::Vector{C}
+end
+            
+
+
+"""   FCollector(depth::Int,frame::Frame)
+
+this is an initialization funtion for the FCollector
+
+"""
+function FCollector(d::Int, f)
+    FCollector(d, f, FCollector[])
+end
+
+""" Frame(func, args, ret, subtrace)
+
+a structure to hold metadata for recursive type information for each function call
+Every frame can be thought of as a single stack frame when a function is called
+            
+"""
+mutable struct Frame{F,T,U}
+    func::F
+    args::T
+    ret::U
+end
+            
+function Cassette.overdub(ctx::TypeCtx, f, args...) # add boilerplate for functionality
```

Comment:
do these need to be defined in the top level script because of world age? I thought this could also be imported from the module. It is just the `Cassette.overdub(ctx, f::typeof(funcname), args...)` that needs to come after the definition of `funcname`